hooks.go

```diff
@@ -24,11 +28,13 @@ func (hooks LevelHooks) Add(hook Hook) {
 // Fire all the hooks for the passed level. Used by `entry.log` to fire
 // appropriate hooks for a log entry.
 func (hooks LevelHooks) Fire(level Level, entry *Entry) error {
+	var aggErr error
+
 	for _, hook := range hooks[level] {
 		if err := hook.Fire(entry); err != nil {
-			return err
+			aggErr = fmt.Errorf("%s | %s", aggErr, err)
```

Comment:
@sirupsen A circuit breaker sounds fine. I've also used https://godoc.org/golang.org/x/time/rate successfully, but I am not sure if this warrants the dependency. You might be able to crib from [here](https://github.com/docker/go-events/blob/master/retry.go#L126) (don't use that as a dependency, as we may change the interface).

Another possibility would be to increment and export a counter through expvar or prometheus, but that might be more involved.